### DISCUSSION TITLE
[Mailer]Mailer object sending envelope with null?

### DISCUSSION THREAD

**Discussion by Melek-E** [symfony/symfony#46556] [Q&A] [answered]
### Symfony version(s) affected

5.4

### Description

Mailer throwing an error on sign-up of a new user, user gets added to the database but the email doesn't get sent.
Error:
`
An exception occurred while executing a query: SQLSTATE[23502]: Not null violation: 7 ERROR: null value in column "id" violates not-null constraint
DETAIL: Failing row contains (null, O:36:\"Symfony\\Component\\Messenger\\Envelope\":2:{s:44:\"\0Sym..., [], default, 2022-06-02 10:35:43, 2022-06-02 10:35:43, null).
Log:
`
( 2022-06-02 10:35:43 is the timestamp at the time of the request)
`
<html><body>

INFO                 10:35:41 | request | Matched route "app_register".                                {     "route": "app_register",     "route_parameters": {         "_route": "app_register",         "_controller": "App\\Controller\\RegistrationController::register"     },     "request_uri": "https://127.0.0.1:8000/register",     "method": "POST" }
-- | -- | --
DEBUG                 10:35:41 | security | Checking for authenticator support.                                {     "firewall_name": "main",     "authenticators": 1 }
DEBUG                 10:35:41 | security | Checking support on authenticator.                                {     "firewall_name": "main",     "authenticator": "App\\Security\\Authenticator" }
DEBUG                 10:35:41 | security | Authenticator does not support the request.                                {     "firewall_name": "main",     "authenticator": "App\\Security\\Authenticator" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\DebugHandlersListener::configure".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\DebugHandlersListener::configure" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\ValidateRequestListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\ValidateRequestListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\SessionListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\SessionListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\LocaleListener::setDefaultLocale".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\LocaleListener::setDefaultLocale" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "FOS\RestBundle\EventListener\FormatListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "FOS\\RestBundle\\EventListener\\FormatListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\RouterListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\RouterListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\LocaleListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\LocaleListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\LocaleAwareListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\LocaleAwareListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener::configureLogoutUrlGenerator".                                 {     "event": "kernel.request",     "listener": "Symfony\\Bundle\\SecurityBundle\\Debug\\TraceableFirewallListener::configureLogoutUrlGenerator" }
DEBUG                 10:35:41 | event | Notified event "kernel.request" to listener  "Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Bundle\\SecurityBundle\\Debug\\TraceableFirewallListener::onKernelRequest" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "FOS\RestBundle\EventListener\ParamFetcherListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "FOS\\RestBundle\\EventListener\\ParamFetcherListener::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "Symfony\Bundle\FrameworkBundle\DataCollector\RouterDataCollector::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Symfony\\Bundle\\FrameworkBundle\\DataCollector\\RouterDataCollector::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "Symfony\Component\HttpKernel\DataCollector\RequestDataCollector::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Symfony\\Component\\HttpKernel\\DataCollector\\RequestDataCollector::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\ControllerListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\ControllerListener::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\ParamConverterListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\ParamConverterListener::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\HttpCacheListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\HttpCacheListener::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\TemplateListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\TemplateListener::onKernelController" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller_arguments" to listener   "Symfony\Component\HttpKernel\EventListener\ErrorListener::onControllerArguments".                                 {     "event": "kernel.controller_arguments",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\ErrorListener::onControllerArguments" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller_arguments" to listener   "Sensio\Bundle\FrameworkExtraBundle\EventListener\SecurityListener::onKernelControllerArguments".                                 {     "event": "kernel.controller_arguments",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\SecurityListener::onKernelControllerArguments" }
DEBUG                 10:35:41 | event | Notified event "kernel.controller_arguments" to listener   "Sensio\Bundle\FrameworkExtraBundle\EventListener\IsGrantedListener::onKernelControllerArguments".                                 {     "event": "kernel.controller_arguments",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\IsGrantedListener::onKernelControllerArguments" }
INFO                 10:35:42 | doctrine | Connecting with parameters {params}                                {     "params": {         "url": "<redacted>",         "driver": "pdo_pgsql",         "host": "137.74.219.115",         "port": 5432,         "user": "remorka",         "password": "<redacted>",         "driverOptions": [],         "defaultTableOptions": [],         "dbname": "remorka",         "serverVersion": "13",         "charset": "utf8"     } }
DEBUG                 10:35:42 | doctrine | Executing statement: SELECT t0.id AS id_1, t0.email AS  email_2, t0.roles AS roles_3, t0.password AS password_4, t0.is_verified  AS is_verified_5 FROM "user" t0 WHERE t0.email = ? LIMIT 2 (parameters:  {params}, types: {types})                                {     "sql": "SELECT t0.id AS id_1, t0.email AS email_2, t0.roles AS roles_3, t0.password AS password_4, t0.is_verified AS is_verified_5 FROM \"user\" t0 WHERE t0.email = ? LIMIT 2",     "params": {         "1": "melek-essid@live.fr"     },     "types": {         "1": 2     } }
DEBUG                 10:35:43 | doctrine | Executing query: SELECT NEXTVAL('user_id_seq')                                {     "sql": "SELECT NEXTVAL('user_id_seq')" }
DEBUG                 10:35:43 | doctrine | Beginning transaction
DEBUG                 10:35:43 | doctrine | Executing statement: INSERT INTO "user" (id, email,  roles, password, is_verified) VALUES (?, ?, ?, ?, ?) (parameters:  {params}, types: {types})                                {     "sql": "INSERT INTO \"user\" (id, email, roles, password, is_verified) VALUES (?, ?, ?, ?, ?)",     "params": {         "1": 24,         "2": "melek-essid@live.fr",         "3": "[]",         "4": "$2y$13$/bLTMdE2EbNJ5YyrPZA8QeMMzexbWLnY3UCic6uUfklJppF17JVCu",         "5": 0     },     "types": {         "1": 1,         "2": 2,         "3": 2,         "4": 2,         "5": 5     } }
DEBUG                 10:35:43 | doctrine | Committing transaction
DEBUG                 10:35:43 | event | Notified event  "Symfony\Component\Mailer\Event\MessageEvent" to listener  "Symfony\Component\Mailer\EventListener\MessageListener::onMessage".                                 {     "event": "Symfony\\Component\\Mailer\\Event\\MessageEvent",     "listener": "Symfony\\Component\\Mailer\\EventListener\\MessageListener::onMessage" }
DEBUG                 10:35:43 | event | Notified event  "Symfony\Component\Mailer\Event\MessageEvent" to listener  "Symfony\Component\Mailer\EventListener\EnvelopeListener::onMessage".                                 {     "event": "Symfony\\Component\\Mailer\\Event\\MessageEvent",     "listener": "Symfony\\Component\\Mailer\\EventListener\\EnvelopeListener::onMessage" }
DEBUG                 10:35:43 | event | Notified event  "Symfony\Component\Mailer\Event\MessageEvent" to listener  "Symfony\Component\Mailer\EventListener\MessageLoggerListener::onMessage".                                 {     "event": "Symfony\\Component\\Mailer\\Event\\MessageEvent",     "listener": "Symfony\\Component\\Mailer\\EventListener\\MessageLoggerListener::onMessage" }
INFO                 10:35:43 | messenger | Sending message  Symfony\Component\Mailer\Messenger\SendEmailMessage with async sender  using  Symfony\Component\Messenger\Bridge\Doctrine\Transport\DoctrineTransport                                 {     "message": {},     "class": "Symfony\\Component\\Mailer\\Messenger\\SendEmailMessage",     "alias": "async",     "sender": "Symfony\\Component\\Messenger\\Bridge\\Doctrine\\Transport\\DoctrineTransport" }
DEBUG                 10:35:43 | doctrine | Executing statement: INSERT INTO messenger_messages  (body, headers, queue_name, created_at, available_at) VALUES(?, ?, ?, ?,  ?) (parameters: {params}, types: {types})                                 {     "sql": "INSERT INTO messenger_messages (body, headers, queue_name, created_at, available_at) VALUES(?, ?, ?, ?, ?)",     "params": {         "1": "O:36:\\\"Symfony\\\\Component\\\\Messenger\\\\Envelope\\\":2:{s:44:\\\"\\0Symfony\\\\Component\\\\Messenger\\\\Envelope\\0stamps\\\";a:1:{s:46:\\\"Symfony\\\\Component\\\\Messenger\\\\Stamp\\\\BusNameStamp\\\";a:1:{i:0;O:46:\\\"Symfony\\\\Component\\\\Messenger\\\\Stamp\\\\BusNameStamp\\\":1:{s:55:\\\"\\0Symfony\\\\Component\\\\Messenger\\\\Stamp\\\\BusNameStamp\\0busName\\\";s:21:\\\"messenger.bus.default\\\";}}}s:45:\\\"\\0Symfony\\\\Component\\\\Messenger\\\\Envelope\\0message\\\";O:51:\\\"Symfony\\\\Component\\\\Mailer\\\\Messenger\\\\SendEmailMessage\\\":2:{s:60:\\\"\\0Symfony\\\\Component\\\\Mailer\\\\Messenger\\\\SendEmailMessage\\0message\\\";O:28:\\\"Symfony\\\\Component\\\\Mime\\\\Email\\\":6:{i:0;N;i:1;N;i:2;s:23:\\\"<p>Nice to meet you</p>\\\";i:3;s:5:\\\"utf-8\\\";i:4;a:0:{}i:5;a:2:{i:0;O:37:\\\"Symfony\\\\Component\\\\Mime\\\\Header\\\\Headers\\\":2:{s:46:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\Headers\\0headers\\\";a:3:{s:4:\\\"from\\\";a:1:{i:0;O:47:\\\"Symfony\\\\Component\\\\Mime\\\\Header\\\\MailboxListHeader\\\":5:{s:50:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0name\\\";s:4:\\\"From\\\";s:56:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0lineLength\\\";i:76;s:50:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0lang\\\";N;s:53:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0charset\\\";s:5:\\\"utf-8\\\";s:58:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\MailboxListHeader\\0addresses\\\";a:1:{i:0;O:30:\\\"Symfony\\\\Component\\\\Mime\\\\Address\\\":2:{s:39:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Address\\0address\\\";s:24:\\\"weirdfoodstuff@gmail.com\\\";s:36:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Address\\0name\\\";s:0:\\\"\\\";}}}}s:2:\\\"to\\\";a:1:{i:0;O:47:\\\"Symfony\\\\Component\\\\Mime\\\\Header\\\\MailboxListHeader\\\":5:{s:50:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0name\\\";s:2:\\\"To\\\";s:56:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0lineLength\\\";i:76;s:50:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0lang\\\";N;s:53:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0charset\\\";s:5:\\\"utf-8\\\";s:58:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\MailboxListHeader\\0addresses\\\";a:1:{i:0;O:30:\\\"Symfony\\\\Component\\\\Mime\\\\Address\\\":2:{s:39:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Address\\0address\\\";s:15:\\\"ahmed@gmail.com\\\";s:36:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Address\\0name\\\";s:0:\\\"\\\";}}}}s:7:\\\"subject\\\";a:1:{i:0;O:48:\\\"Symfony\\\\Component\\\\Mime\\\\Header\\\\UnstructuredHeader\\\":5:{s:50:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0name\\\";s:7:\\\"Subject\\\";s:56:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0lineLength\\\";i:76;s:50:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0lang\\\";N;s:53:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\AbstractHeader\\0charset\\\";s:5:\\\"utf-8\\\";s:55:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\UnstructuredHeader\\0value\\\";s:25:\\\"Welcome to the Space Bar!\\\";}}}s:49:\\\"\\0Symfony\\\\Component\\\\Mime\\\\Header\\\\Headers\\0lineLength\\\";i:76;}i:1;N;}}s:61:\\\"\\0Symfony\\\\Component\\\\Mailer\\\\Messenger\\\\SendEmailMessage\\0envelope\\\";N;}}",         "2": "[]",         "3": "default",         "4": "2022-06-02 10:35:43",         "5": "2022-06-02 10:35:43"     },     "types": {         "1": 2,         "2": 2,         "3": 2,         "4": 2,         "5": 2     } }
CRITICAL                 10:35:43 | request | Uncaught PHP Exception  Symfony\Component\Messenger\Exception\TransportException: "An exception  occurred while executing a query: SQLSTATE[23502]: Not null violation: 7  ERROR:  null value in column "id" violates not-null constraint DETAIL:  Failing row contains (null,  O:36:\"Symfony\\Component\\Messenger\\Envelope\":2:{s:44:\"\0Sym..., [],  default, 2022-06-02 10:35:43, 2022-06-02 10:35:43, null)." at  C:\Users\Monke\number3\remorka\vendor\symfony\doctrine-messenger\Transport\DoctrineSender.php  line 51                                {     "exception": {} }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\DebugHandlersListener::configure".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\DebugHandlersListener::configure" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\ValidateRequestListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\ValidateRequestListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\SessionListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\SessionListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\LocaleListener::setDefaultLocale".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\LocaleListener::setDefaultLocale" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "FOS\RestBundle\EventListener\FormatListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "FOS\\RestBundle\\EventListener\\FormatListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\RouterListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\RouterListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\LocaleListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\LocaleListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Component\HttpKernel\EventListener\LocaleAwareListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\LocaleAwareListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener::configureLogoutUrlGenerator".                                 {     "event": "kernel.request",     "listener": "Symfony\\Bundle\\SecurityBundle\\Debug\\TraceableFirewallListener::configureLogoutUrlGenerator" }
DEBUG                 10:35:43 | event | Notified event "kernel.request" to listener  "Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener::onKernelRequest".                                 {     "event": "kernel.request",     "listener": "Symfony\\Bundle\\SecurityBundle\\Debug\\TraceableFirewallListener::onKernelRequest" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "FOS\RestBundle\EventListener\ParamFetcherListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "FOS\\RestBundle\\EventListener\\ParamFetcherListener::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "Symfony\Bundle\FrameworkBundle\DataCollector\RouterDataCollector::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Symfony\\Bundle\\FrameworkBundle\\DataCollector\\RouterDataCollector::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "Symfony\Component\HttpKernel\DataCollector\RequestDataCollector::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Symfony\\Component\\HttpKernel\\DataCollector\\RequestDataCollector::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\ControllerListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\ControllerListener::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\ParamConverterListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\ParamConverterListener::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\HttpCacheListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\HttpCacheListener::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller" to listener  "Sensio\Bundle\FrameworkExtraBundle\EventListener\TemplateListener::onKernelController".                                 {     "event": "kernel.controller",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\TemplateListener::onKernelController" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller_arguments" to listener   "Symfony\Component\HttpKernel\EventListener\ErrorListener::onControllerArguments".                                 {     "event": "kernel.controller_arguments",     "listener": "Symfony\\Component\\HttpKernel\\EventListener\\ErrorListener::onControllerArguments" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller_arguments" to listener   "Sensio\Bundle\FrameworkExtraBundle\EventListener\SecurityListener::onKernelControllerArguments".                                 {     "event": "kernel.controller_arguments",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\SecurityListener::onKernelControllerArguments" }
DEBUG                 10:35:43 | event | Notified event "kernel.controller_arguments" to listener   "Sensio\Bundle\FrameworkExtraBundle\EventListener\IsGrantedListener::onKernelControllerArguments".                                 {     "event": "kernel.controller_arguments",     "listener": "Sensio\\Bundle\\FrameworkExtraBundle\\EventListener\\IsGrantedListener::onKernelControllerArguments" }

</body>
</html>
`

### How to reproduce

I'm following this tutorial for [symfonycasts](https://symfonycasts.com/screencast/mailer/email-object)
To replicate this error: create a registration form that sends a new signed up user to the database, create email and mailer objects.
`
<?php

namespace App\Controller;
use App\Entity\User;
use App\Form\RegistrationFormType;
use Doctrine\ORM\EntityManagerInterface;
use Symfony\Bridge\Twig\Mime\TemplatedEmail;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Mime\Address;
use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Contracts\Translation\TranslatorInterface;
use SymfonyCasts\Bundle\VerifyEmail\Exception\VerifyEmailExceptionInterface;
use Symfony\Component\Mime\Email;
use Symfony\Component\Mailer\MailerInterface;
class RegistrationController extends AbstractController
{
    
    #[Route('/register', name: 'app_register')]
    public function register(MailerInterface $mailer, Request $request, UserPasswordHasherInterface $userPasswordHasher, EntityManagerInterface $entityManager): Response
    {   $randomNumber = rand();
        $user = new User($randomNumber);
        $form = $this->createForm(RegistrationFormType::class, $user);
        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid())
        {
            // encode the plain password
            $user->setPassword(
            $userPasswordHasher->hashPassword(
                    $user,
                    $form->get('plainPassword')->getData()
                )
            );

            $entityManager->persist($user);
            $entityManager->flush();

            $email = (new Email())
                ->from('testing@gmail.com')
                ->to('melek@gmail.com')
                ->subject('Welcome!')
                ->html("<p>Nice to meet you</p>");
            $mailer->send($email);
            return $this->redirectToRoute('api_app_remorque_list');
            
        }

        return $this->render('registration/register.html.twig', [
            'registrationForm' => $form->createView(),
        ]);
    }

}
`

(The   return $this->render('registration/register.html.twig', [  'registrationForm' => $form->createView(), only goes through if I delete or comment the mailer line).

### Additional Context

![image](https://user-images.githubusercontent.com/86518925/171614376-682c0ff4-efcb-4ac4-9871-b81de61b01f4.png)
SMTP:
MAILER_DSN is set up correctly using mailtrap.

**Comment by dbrumann** [comment]
It looks like this is an issue with the Messenger not the Mailer. It looks like you are dispatching Emails via Messenger and storing the message in the doctrine transport fails. I can't see how this can happen. How did you set up the messenger? Did you run `bin/console messenger:setup-transports` to create the table?

**Reply by Melek-E** [reply]
I did not use bin/console messenger:setup-transports, the table was created using doctrine/php bin/console make:user from the MakerBundle. 
as for the messenger:
config/packages/messenger.yaml
`framework:
    messenger:
        failure_transport: failed

        transports:
            # https://symfony.com/doc/current/messenger.html#transport-configuration
            async:
                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
                options:
                    use_notify: true
                    check_delayed_interval: 60000
                retry_strategy:
                    max_retries: 3
                    multiplier: 2
            failed: 'doctrine://default?queue_name=failed'
            # sync: 'sync://'

        routing:
            Symfony\Component\Mailer\Messenger\SendEmailMessage: async
            Symfony\Component\Notifier\Message\ChatMessage: async
            Symfony\Component\Notifier\Message\SmsMessage: async
            
            # Route your messages to the transports
            # 'App\Message\YourMessage': async

`

```
.env
`###> symfony/messenger ###
# Choose one of the transports below
MESSENGER_TRANSPORT_DSN=doctrine://default
# MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages
# MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages
###< symfony/messenger ###`

```

**Comment by dbrumann** [comment, marked as answer]
Try changing your `config/packages/messenger.yaml` to this configuration:

```
        sync: 'sync://'

routing:
        Symfony\Component\Mailer\Messenger\SendEmailMessage: sync
        Symfony\Component\Notifier\Message\ChatMessage: sync
        Symfony\Component\Notifier\Message\SmsMessage: sync
```

Afterwards your code should work, but these messages will no longer be queued for delivery and instead will be sent straight away. If you want to re-enable the feature try removing the existing messenger-table in your database and then recreating it with the `messenger:setup-transports`-command I mentioned before. If that does not solve your issue, please show the output of the command (preferably with verbose output enabled).

**Reply by Melek-E** [reply]
Thank you for the help, that was it!
